### PR TITLE
Add Schema for YAML extensions

### PR DESCRIPTION
**Pull request by jacques-n**
- add schema for extensions
- update existing extensions to validate against schema
- add an example "unknown" extension to illustrate simplified possibilities.
- add github workflow to validate yaml files.

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -34,10 +34,27 @@ jobs:
...
+        with:
+          node-version: '14'
+      - run: npm install -g ajv-cli
+      - run: |
+          set -e
+          for i in $(ls);
+          do
+            ajv validate -s ../text/simple_extensions_schema.yaml --strict=true --spec=draft2020 -d $i
+          done
+          set +e
```
Comment (cpcloud) [inline .github/workflows/pr.yml:59]: This doesn't seem necessary. It's fine to leave `-e` enabled.

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -34,10 +34,27 @@ jobs:
...
+    name: Validate YAML extensions
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '14'
+      - run: npm install -g ajv-cli
+      - run: |
+          set -e
```
Comment (cpcloud) [inline .github/workflows/pr.yml:54]: Should probably just do the usual `set -euo pipefail`

File: .github/workflows/pr.yml [hunk 3]
```diff
@@ -34,10 +34,27 @@ jobs:
...
+          for i in $(ls);
+          do
+            ajv validate -s ../text/simple_extensions_schema.yaml --strict=true --spec=draft2020 -d $i
```
Comment (cpcloud) [inline .github/workflows/pr.yml:57]: Quote `$i`

File: extensions/unknown.yaml [hunk 4]
```diff
@@ -0,0 +1,71 @@
+%YAML 1.2
+---
+types:
+  - name: unknown
+scalar_functions:
+  -
```
Comment (cpcloud) [inline extensions/unknown.yaml:6]: Why the extra newline?

File: .github/workflows/pr.yml (hunk 2 above)
Comment (jacques-n) [reply .github/workflows/pr.yml]: done. honestly just copying pattern from above :)

File: .github/workflows/pr.yml (hunk 1 above)
Comment (jacques-n) [reply .github/workflows/pr.yml]: removed.

File: extensions/unknown.yaml (hunk 4 above)
Comment (jacques-n) [reply extensions/unknown.yaml]: Removed, was a style choice that I don't care much about. The other extensions use this to make it easier to read.

**Review by cpcloud — APPROVED** [review]

**Comment by cpcloud** [comment]
Thanks!

**Pull request merged**